lms/djangoapps/course_api/fields.py

```diff
@@ -0,0 +1,26 @@
+""" Custom fields for Django REST Framework. """
+
+from __future__ import unicode_literals
+
+from rest_framework import fields
+
+
+class AbsoluteURLField(fields.URLField):
+    """
+    Field that serializes values to absolute URLs based on the current request.
+
+    If the value to be serialized is already a URL, that value will returned.
+    """
+
+    def to_representation(self, value):
+        request = self.context.get('request', None)
+
+        assert request is not None, (
```

Comment:
Generally I agree with you. In this case, if we are using `Viewsets` (which we should always be doing), the serializers will always have a request, so I'm less worried about this error actually being raised than other code where I, too, have requested a custom exception class.